service/handlers.go

```diff
@@ -321,7 +321,8 @@ func (h handler) createWorkflowFromRequest(_ context.Context, w http.ResponseWri
 		return
 	}
 	if !targetExists {
-		h.errorResponse(w, fmt.Sprintf("target '%s' does not exist for project `%s`", cwr.TargetName, cwr.ProjectName), http.StatusBadRequest)
+		level.Error(l).Log("message", "target not found")
+		h.errorResponse(w, "target not found", http.StatusBadRequest)
```

Comment:
It's probably from existing inconsistencies, but I think this should be 404

Reply:
Scratch that. This is part of the workflow request, so makes sense to be a bad request.